examples/systemcalls/systemcalls.c

```diff
@@ -92,8 +145,65 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
-
+    
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if(fd < 0)
+    {
+    	perror("open");
+    	return false;
+    }
+    
+    pid_t childPid = fork();
+    int waitStatus;
+    
+    switch(childPid)
+    {
+    
+    case -1:
+    	
+    	perror("fork");
+    	close(fd);
+    	return false;
+    	
+    case 0:
+    
+    	// Inside child
+    	if(dup2(fd, 1) < 0)
+    	{
+    		perror("dup2");
+    		abort();
+    	}
+    	close(fd);
+    	
+    	// Execute the command using execv()
+    	if(execv(command[0], command) == -1)
+    	{
+    		abort();
+    	}
+    	
+    default:
+    
+    	// Inside parent
+    	waitpid(childPid, &waitStatus, 0);
+    	if(waitStatus == -1) // Error
+    	{
+    		return false;
+    	}
+    	
+    	// Check if the child process exited correctly with zero return code
+    	if(WIFEXITED(waitStatus) == true)
+    	{
+    		if(WEXITSTATUS(waitStatus) != 0)
+    		{
+    			return false;
+    		}
+    	}
+    	else
+    	{
+    		return false;
+    	}
+    }
```

Comment:
@suryakanteti think about how you could avoid duplicating logic between these two do_exec functions.